### PR TITLE
Add support for Neovim-native LSP

### PR DESCRIPTION
**Pull request by dradtke**
Please use the following structure when proposing changes to our shared Vim configuration and make sure to complete the checklist at the end.

# What

Neovim 0.5 has support for native LSP. This PR detects when the user is running Neovim 0.5, and configures the editor's native LSP over `vim-lsp`.

# Why

So that users interested in using Neovim's native LSP support can do so.

# Checklist

- [ ] Send RFC email to Braintree developers _if change may be impactful_. Please include a link to this PR so that discussion about the pros and cons of the change remains linked to the proposed changes. **Recent examples include**: addition of linter and completer, no longer removing end-of-line whitespace on save, changing to fzf for file lookup.

### CODE REVIEW

**Review by jaresty — APPROVED** [review]

**Comment by dradtke** [comment]
Apologies for the noise, I was making a couple improvements + updates to the documentation as long as this wasn't merged, but this should be good to go anytime if it looks good to others.

File: config/nvim/init.vim [hunk 1]
```diff
@@ -1,3 +1,93 @@
     set runtimepath^=~/.vim runtimepath+=~/.vim/after
     let &packpath = &runtimepath
     source ~/.vim/vimrc
+
+if has('nvim-0.5')
```
Comment (ClashTheBunny) [inline config/nvim/init.vim:5]: Does this do >0.5, like the normal vim "contains feature number" thing?

**Review by ClashTheBunny — APPROVED** [review]
This looks like it will work!

File: config/nvim/init.vim (hunk 1 above)
Comment (dradtke) [reply config/nvim/init.vim:5]: Yep, it should work for Neovim 0.5 and everything later.

**Comment by dradtke** [comment]
Just updated to resolve merge conflicts

**Pull request merged**